### PR TITLE
Use fixedthreadpool instead of forkjoinpool for GPRC

### PR DESCRIPTION
**Pull request by yuzhu**

### CODE REVIEW

**Comment by yuzhu** [comment]
@calvinjia changing to fixedthreadpool with default 500 max threads.

File: core/server/master/src/main/java/alluxio/master/AlluxioMasterProcess.java [hunk 1]
```diff
@@ -280,9 +280,9 @@ protected void startServingRPCServer() {
       GrpcServerBuilder serverBuilder = GrpcServerBuilder.forAddress(mRpcBindAddress,
           ServerConfiguration.global());
 
-      mRPCExecutor =
-          new ForkJoinPool(ServerConfiguration.getInt(
-              PropertyKey.MASTER_RPC_FORKJOIN_POOL_PARALLELISM));
+      mRPCExecutor = Executors.newFixedThreadPool(ServerConfiguration.getInt(
```
Comment (ggezer) [inline core/server/master/src/main/java/alluxio/master/AlluxioMasterProcess.java]: You can use mMinWorkerThreads/mMaxWorkerThreads member variables of the base MasterProcess. 

Default supplier for MASTER_WORKER_THREADS_MAX key will need to be modified though, it currently returns around 30K in linux. We don't want a high upper bound with gRPC.

File: core/server/master/src/main/java/alluxio/master/AlluxioMasterProcess.java [hunk 2]
```diff
@@ -280,9 +280,9 @@ protected void startServingRPCServer() {
...
+          PropertyKey.MASTER_RPC_THREAD_POOL_PARALLELISM));
```
Comment (calvinjia) [inline core/server/master/src/main/java/alluxio/master/AlluxioMasterProcess.java]: Does it make sense to supply a thread factory?

**Comment by gpang** [comment]
@yuzhu Does this mean this will have the same limitations as the thrift thread pools?

File: core/common/src/main/java/alluxio/conf/PropertyKey.java [hunk 3]
```diff
@@ -3802,8 +3802,8 @@ private static String javadocLink(String fullyQualifiedClassname) {
     public static final String MASTER_JOURNAL_TEMPORARY_FILE_GC_THRESHOLD_MS =
         "alluxio.master.journal.temporary.file.gc.threshold";
 
-    public static final String MASTER_RPC_FORKJOIN_POOL_PARALLELISM =
-        "alluxio.master.rpc.fork.parallelism";
+    public static final String MASTER_RPC_THREAD_POOL_PARALLELISM =
+        "alluxio.master.rpc.threadpool.parallelism";
```
Comment (apc999) [inline core/common/src/main/java/alluxio/conf/PropertyKey.java]: nit: why  `alluxio.master.rpc.threadpool.parallelism` not `alluxio.master.rpc.threadpool.size` which seems easier to remember and type?

File: core/server/master/src/main/java/alluxio/master/AlluxioMasterProcess.java (hunk 1 above)
Comment (yuzhu) [reply core/server/master/src/main/java/alluxio/master/AlluxioMasterProcess.java]: Looks like the default supplier for `MASTER_WORKER_THREADS_MAX` returns 2048 or max file descriptor/3, should we just set it to a constant like 500?

File: core/server/master/src/main/java/alluxio/master/AlluxioMasterProcess.java (hunk 2 above)
Comment (yuzhu) [reply core/server/master/src/main/java/alluxio/master/AlluxioMasterProcess.java]: Maybe? what do we want to do in a custom thread factory?

**Comment by yuzhu** [comment]
@gpang I am not sure what limitation thrift thread pool had? could you give more context?

File: core/server/master/src/main/java/alluxio/master/AlluxioMasterProcess.java (hunk 2 above)
Comment (calvinjia) [reply core/server/master/src/main/java/alluxio/master/AlluxioMasterProcess.java]: We can name the threads and make them daemon threads.

File: core/server/master/src/main/java/alluxio/master/AlluxioMasterProcess.java [hunk 4]
```diff
@@ -281,7 +282,7 @@ protected void startServingRPCServer() {
           ServerConfiguration.global());
 
       mRPCExecutor = Executors.newFixedThreadPool(ServerConfiguration.getInt(
-          PropertyKey.MASTER_RPC_THREAD_POOL_PARALLELISM));
+          PropertyKey.MASTER_RPC_THREAD_POOL_SIZE), new GrpcThreadFactory());
```
Comment (calvinjia) [inline core/server/master/src/main/java/alluxio/master/AlluxioMasterProcess.java]: Use `ThreadFactoryUtils`

File: core/server/master/src/main/java/alluxio/master/AlluxioMasterProcess.java (hunk 1 above)
Comment (ggezer) [reply core/server/master/src/main/java/alluxio/master/AlluxioMasterProcess.java]: For now it's fine to set it to a constant. We would eventually need to come up with a value that is calculated with core-count/memory into account. That'll need further profiling with various machine configurations.

Comment (ggezer) [reply core/server/master/src/main/java/alluxio/master/AlluxioMasterProcess.java]: The reason for this limitation being; 
(max file descriptor/3) returns a very big number(~30K) and executors don't cool off as they used to with thrift, so thread count ends up rapidly climbing up to a high bound.

File: core/common/src/main/java/alluxio/conf/PropertyKey.java [hunk 5]
```diff
@@ -1712,9 +1711,9 @@ public String toString() {
           .setConsistencyCheckLevel(ConsistencyCheckLevel.WARN)
           .setScope(Scope.MASTER)
           .build();
-  public static final PropertyKey MASTER_RPC_FORKJOIN_POOL_PARALLELISM =
-      new Builder(Name.MASTER_RPC_FORKJOIN_POOL_PARALLELISM)
-          .setDefaultValue(50)
+  public static final PropertyKey MASTER_RPC_THREAD_POOL_SIZE =
```
Comment (ggezer) [inline core/common/src/main/java/alluxio/conf/PropertyKey.java]: This had left over?

Comment (yuzhu) [reply core/common/src/main/java/alluxio/conf/PropertyKey.java]: I will make a PR to remove it.

**Comment by ggezer** [comment]
@yuzhu
Now that the master is not using FJP, I believe we should revert the client side code to use common FJP as well...

**Comment by yuzhu** [comment]
@ggezer I think @bf8086 already made that change to use common FJP. See #8438

**Comment by ggezer** [comment]
@yuzhu #8438 was to make data path clients not to use FJP common pool. Rest are still using the FJP common pool.

**Pull request merged**